File: backend/src/strata_ai/schemas/organization.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict
from pydantic.alias_generators import to_camel

from strata_ai.models import OrganizationKind


class OrganizationCreate(BaseModel):
    model_config = ConfigDict(extra="forbid", alias_generator=to_camel)

    name: str
    kind: OrganizationKind
    industry: str
    is_active: bool = True


class OrganizationUpdate(BaseModel):
    model_config = ConfigDict(extra="forbid", alias_generator=to_camel)

    name: Optional[str] = None
    kind: Optional[OrganizationKind] = None
    industry: Optional[str] = None


class OrganizationRead(BaseModel):
    model_config = ConfigDict(
        from_attributes=True,
        extra="forbid",
        alias_generator=to_camel,
        validate_by_name=True,
    )

    name: str
    kind: OrganizationKind
    industry: str
    id: uuid.UUID
    is_active: bool
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/organization.ts
export type OrganizationKind = "client" | "vendor" | "internal";

export type Organization = {
  id: string;
  name: string;
  kind: OrganizationKind;
  industry: string;
  isActive: boolean;
  createdAt: Date;
  updatedAt: Date;
};

export type OrganizationCreateInput = {
  name: string;
  kind: OrganizationKind;
  industry: string;
  isActive?: boolean;
};

export type OrganizationUpdateInput = {
  id: string;
  name?: string;
  kind?: OrganizationKind;
  industry?: string;
  isActive?: boolean;
};

export type OrganizationIdInput = {
  id: string;
};
